sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlParser.g

```diff
@@ -1756,6 +1795,28 @@ showStmtIdentifier
     : identifier
     | StringLiteral
     ;
+tableProvider
+@init { pushMsg("table's provider", state); }
+@after { popMsg(state); }
+    :
+      KW_USING provider=Identifier  -> ^(TOK_TABLEPROVIDER $provider)
```

Comment:
Providers can have dotted names, e.g.: ```org.apache.spark.sql.avro```. Identifier won't work here. You would need something like: ```Identifier (DOT Identifier)*```

I am not sure if we should support quoted identifiers here?